pkg/cleaner/cleaner_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleaner
+
+import (
+	scfake "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/fake"
+	"github.com/kubernetes-incubator/service-catalog/pkg/probe"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/apps/v1beta1"
+	extv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextfake "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	"testing"
+)
+
+const (
+	cmName      = "controller-name"
+	cmNamespace = "relase-namespace"
+)
+
+func TestCleaner_RemoveCRDs(t *testing.T) {
+	// Given
+	fakeClik8s := k8sfake.NewSimpleClientset(newTestDeployment())
+	fakeCliext := apiextfake.NewSimpleClientset(newTestCRD()...)
+	fakeClisc := scfake.NewSimpleClientset()
+
+	clr := New(fakeClik8s, fakeClisc, fakeCliext)
+
+	// When
+	assert.NoError(t, clr.RemoveCRDs(cmNamespace, cmName))
+
+	// Then
+	list, err := fakeCliext.ApiextensionsV1beta1().CustomResourceDefinitions().List(v1.ListOptions{})
+	assert.NoError(t, err)
+	// one of the CRD is not connected with ServiceCatalog
+	assert.Len(t, list.Items, 1)
```

Comment:
can w also assert the name? then the comment above will not be necessary and code will be self-described